### ISSUE TITLE
ThomasHaile: Week 4, Web Apps

### ISSUE THREAD

**Issue by thomashaile** [HackYourFutureBelgium/class-8#358]
### Prep Work

- [ ] one prep topic
- [ ] the next prep topic
- [ ] ...

### Name of Project:

- [repository](github.com/user-name/repo-name)
- [live demo](https://user-name.github.io/repo-name)
- [ ]  [step 1](link to branch for version 1)
- [ ]  [step 2](link to branch for version 2)
- [ ]  [step 3](link to branch for version 3)
- [ ]  ...

### Exercises:

- [ ] exercise name: [repo](https://github.com/user-name/repo-name), [live (if there is)]()
- [ ] exercise name: [repo](https://github.com/user-name/repo-name), [live (if there is)]()
- [ ] ...

> [Follow this link](https://github.com/HackYourFutureBelgium/class-6/issues/59) to study a well-done homework issue.

**Comment by thomashaile** [comment]
Done- exercises(schema,data,handler), texidor-validated, and podex-api logic
===
My Next task:- will be pokedex-api handler functions and the project...

**Comment by Toinne** [comment]
💯 
No remarks on the `textidor-validated` looks great.

For the `user-authntication` `repository`

```js
        const data = await fetch('/api/users/' + user.id, {
            method: 'GET',
            body: JSON.stringify(user),
            headers: {
                'Content-type': 'application/json; charset=UTF-8',
            },
        });
```
https://github.com/thomashaile/User-login-node/blob/server-setup/client/app/actions.js#L4

When we want to log in a user we will want to do this based on the identifiers available.
The 2 fields from the form `email` and `password`. So a route in our api call `login` is more appropriate that route should take 2 arguments `email` and `password` and search the user based on that information.

Did not really dig into what the pokeapi exercise.
Small tip here
```js
const id = (pokeData, id) => {
    const pokeD = pokeData.find(element => element.id == id);
    if (pokeD) {
        return pokeD;
    } else {
        return null;
    }
}	
```
No need to write an else 

```
const id = (pokeData, id) => {
    const pokeD = pokeData.find(element => element.id == id);
    if (pokeD) {
        return pokeD;
    }
 
    return null;
}
```
Makes it shorter and less conditionals to worry about.

**Comment by thomashaile** [comment]
Yes.... anthony thank you i am stacked there.. i will try it based on your
comment.
